Makefile

```diff
@@ -1,321 +1,259 @@
+SCORECARD_TEST_IMAGE ?= $(SCORECARD_TEST_CI_IMAGE)
+SCORECARD_TEST_ARCHES:="amd64" "ppc64le" "arm64" "s390x"
 image-build-scorecard-test:
-	./hack/image/build-scorecard-test-image.sh $(SCORECARD_TEST_BASE_IMAGE):dev
-
+	./hack/image/build-scorecard-test-image.sh $(SCORECARD_TEST_CI_IMAGE):dev
 image-push-scorecard-test:
-	./hack/image/push-image-tags.sh $(SCORECARD_TEST_BASE_IMAGE):dev $(SCORECARD_TEST_IMAGE)-$(shell go env GOARCH)
-
+	./hack/image/push-image-tags.sh $(SCORECARD_TEST_CI_IMAGE):dev $(SCORECARD_TEST_IMAGE)-$(shell go env GOARCH)
 image-push-scorecard-test-multiarch:
 	./hack/image/push-manifest-list.sh $(SCORECARD_TEST_IMAGE) ${SCORECARD_TEST_ARCHES}
 
-# Scorecard test kuttl image build/push.
 .PHONY: image-build-scorecard-test-kuttl image-push-scorecard-test-kuttl image-push-scorecard-test-kuttl-multiarch
-
+SCORECARD_TEST_KUTTL_CI_IMAGE = quay.io/operator-framework/scorecard-test-kuttl
+SCORECARD_TEST_KUTTL_IMAGE ?= $(SCORECARD_TEST_KUTTL_CI_IMAGE)
+SCORECARD_TEST_KUTTL_ARCHES:="amd64" "ppc64le" "arm64"
 image-build-scorecard-test-kuttl:
-	./hack/image/build-scorecard-test-kuttl-image.sh $(SCORECARD_TEST_KUTTL_BASE_IMAGE):dev
-
+	./hack/image/build-scorecard-test-kuttl-image.sh $(SCORECARD_TEST_KUTTL_CI_IMAGE):dev
 image-push-scorecard-test-kuttl:
-	./hack/image/push-image-tags.sh $(SCORECARD_TEST_KUTTL_BASE_IMAGE):dev $(SCORECARD_TEST_KUTTL_IMAGE)-$(shell go env GOARCH)
-
+	./hack/image/push-image-tags.sh $(SCORECARD_TEST_KUTTL_CI_IMAGE):dev $(SCORECARD_TEST_KUTTL_IMAGE)-$(shell go env GOARCH)
 image-push-scorecard-test-kuttl-multiarch:
 	./hack/image/push-manifest-list.sh $(SCORECARD_TEST_KUTTL_IMAGE) ${SCORECARD_TEST_KUTTL_ARCHES}
 
-##############################
-# Tests                      #
-##############################
-
-##@ Tests
+.PHONY: image-build-custom-scorecard-tests
+CUSTOM_SCORECARD_TESTS_CI_IMAGE = quay.io/operator-framework/custom-scorecard-tests
+CUSTOM_SCORECARD_TESTS_IMAGE ?= $(CUSTOM_SCORECARD_TESTS_CI_IMAGE)
+image-build-custom-scorecard-tests:
+	./hack/image/build-custom-scorecard-tests-image.sh $(CUSTOM_SCORECARD_TESTS_CI_IMAGE):dev
 
-# Static tests.
-.PHONY: test test-sanity test-unit
+##@ Test
 
-test: test-unit ## Run the tests
+.PHONY: test-all
+test-all: test-static test-e2e ## Run all tests
 
-test-sanity: tidy build/operator-sdk lint
-	./hack/tests/sanity-check.sh
+.PHONY: test-static
+test-static: test-sanity test-unit test-links ## Run all non-cluster-based tests
 
-test-unit: ## Run the unit tests
-	$(Q)go test -coverprofile=coverage.out -covermode=count -count=1 -short $(TEST_PKGS)
+.PHONY: test-sanity
+test-sanity: generate fix ## Test repo formatting, linting, etc.
+	git diff --exit-code # fast-fail if generate or fix produced changes
+	./hack/check-license.sh
+	./hack/check-error-log-msg-format.sh
+	go run ./hack/generate/changelog/gen-changelog.go -validate-only
+	go vet ./...
+	./tools/scripts/fetch golangci-lint 1.31.0 && ./tools/bin/golangci-lint run
```

Comment:
How we can check just the lint?

Reply:
With this PR, I'm intentionally trying to decrease the number of make targets down to a much more manageable set with just the ones that are truly necessary.

IMO `make lint` isn't much easier or better than just directly running `./tools/bin/golangci-lint run`. Personally, I never used `make lint` and always just ran `make test-sanity`.

At the end of the day it comes down to what other contributors want, so if there's more support for keeping `lint` and/or `lint-fix` around, we can.

Reply:
> IMO make lint isn't much easier or better than just directly running ./tools/bin/golangci-lint run. Personally, I never used make lint and always just ran make test-sanity.

IHMO the make lint and make lint-fix are easier to remember than `./tools/bin/golangci-lint run` and `./tools/bin/golangci-lint run --fix` and it would keep the same standard adopted in the KB and operator-lib.

So, I'd vote to keep them since 
- the test-sanity do many checks 
- I have the impression that contributors have more difficult to understand why the sanity logs and know why the test is not working at all. 
- and mainly, because the test-sanity requires that we have all committed to passing on that which make harder work with.

Reply:
My typical approach to submitting PRs is to make all the necessary changes and commits, and then run through the tests locally before submitting the PR.

When I follow this approach, I am effectively running ALL of the same tests that run in CI. If we stick with each individual thing being its own make target, we end up with (IMO) two negative outcomes:
1. There would be so many make targets that contributors don't really know what they should or shouldn't use.
2. Contributors will be less likely to run the full set of CI tests locally before submitting a PR, taking up more CI cycles unnecessarily.

My goal with this PR is to make that process easier and more foolproof. This PR:
1. Decreases the number of targets to make it easier for contributors to know what's what (the make targets now generally align with the phases of the CI run). Now contributors will have less guessing to do to know what to run prior to submitting a PR.
2. Makes the e2e tests run in an isolated kind cluster that is used _only_ for operator-sdk e2e tests. Now contributors can run e2e tests in the same way no matter what their local environment looks like.

IMO the benefits far outweigh the drawbacks.